MAUI/Calendar/customizations.md

```diff
@@ -222,4 +293,252 @@ You can customize the selection cell background and text style in Month, Year, D
+    </calendar:SfCalendar>
+    </Grid>
+
+{% endhighlight %}
+{% highlight c# tabtitle="MonthCellTemplateSelector.cs" %}
+
+public class MonthCellTemplateSelector : DataTemplateSelector
+{
+    public MonthCellTemplateSelector()
+    {
+    }
+    public DataTemplate NormalDateTemplate { get; set; }
+    public DataTemplate TodayDateTemplate { get; set; }
+    public DataTemplate LeadingTrailingDateTemplate { get; set; }
+
+    protected override DataTemplate OnSelectTemplate(object item, BindableObject container)
+    {
+        var monthCellDetails = item as CalendarCellDetails;
+        if (monthCellDetails.Date == DateTime.Today.Date)
+            return TodayDateTemplate;
+        else if (monthCellDetails.IsTrailingOrLeadingDate)
+            return LeadingTrailingDateTemplate;
+        else
+            return NormalDateTemplate;
+    }
+}
+
+{% highlight c# tabtitle="MainPage.xaml.cs" %}
+
+this.Calendar.View = CalendarView.Month;
+
+{% endhighlight %}
+{% endtabs %}
+
+## Year cell appearance using DataTemplate
+
+The year cell appearance can be customized by using the `CellTemplate` property of `YearView` in the `SfCalendar`.
+
+{% tabs %}
+{% highlight xaml tabtitle="MainPage.xaml" %}
+
+<calendar:SfCalendar x:Name="Calendar" View="Decade">
+    <calendar:SfCalendar.YearView>
+        <calendar:CalendarYearView>
+            <calendar:CalendarYearView.CellTemplate>
+                <DataTemplate>
+                    <Grid Background="Pink">
+                        <Label
+                        HorizontalTextAlignment="Center" 
+                        VerticalTextAlignment="Center" 
+                        TextColor="Purple" 
+                        Text="{Binding Date.Year}"/>
+                    </Grid>
+                </DataTemplate>
+            </calendar:CalendarYearView.CellTemplate>
+        </calendar:CalendarYearView>
+    </calendar:SfCalendar.YearView>
+</calendar:SfCalendar>
+
+{% endhighlight %}
+{% highlight c# tabtitle="MainPage.xaml.cs" %}
+
+this.Calendar.YearView = new CalendarYearView()
+{
+    CellTemplate = dataTemplate,
+};
+
+DataTemplate dataTemplate = new DataTemplate(() =>
+{
+    Grid grid = new Grid
+    {
+        Background = Colors.Pink
+    };
+
+    Label label = new Label
+    {
+        HorizontalTextAlignment = TextAlignment.Center,
+        VerticalTextAlignment = TextAlignment.Center,
+        TextColor = Colors.Purple
+    };
+
+    label.SetBinding(Label.TextProperty, new Binding("Date.Year"));
+    grid.Children.Add(label);
+    return grid;
+});
+
+{% endhighlight %}
+{% endtabs %}
+
+## Year cell appearance using DataTemplateSelector
+
+You can customize the month cell appearance by using the `CellTemplate` property of `YearView` in the `SfCalendar`. The `DataTemplateSelector` for a calendar can be used to choose a different data template for each cell in the calendar based on the cell's data. It allows to customize the appearance of a particular cell based on certain conditions.
```

Comment:
You can customize the month cell appearance by using the `CellTemplate` property of `YearView` in the `SfCalendar.` The `DataTemplateSelector` for a calendar can be used to choose a different data template for each cell in the calendar based on the cell's data. It allows customizing the appearance of a particular cell based on certain conditions.

Reply:
Content modified.